packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -276,57 +399,100 @@ export class AzureCommunicationCallAdapter implements CallAdapter {
       }
     }
     return false;
-  }
+  };
 
-  public async mute(): Promise<void> {
+  public mute = async (): Promise<void> => {
     if (!this.call) {
       this.context.setIsLocalMicrophoneEnabled(false);
     }
-    await this.call?.mute();
-  }
 
-  public async unmute(): Promise<void> {
+    return withErrorHandling(
+      this.context,
+      this.emitter,
+      async () => {
+        await this.call?.mute();
+      },
+      Promise.resolve()
+    );
+  };
+
+  public unmute = async (): Promise<void> => {
     if (!this.call) {
       this.context.setIsLocalMicrophoneEnabled(true);
     }
-    await this.call?.unmute();
-  }
 
-  public async startScreenShare(): Promise<void> {
-    if (!this.call?.isScreenSharingOn) await this.handlers.onToggleScreenShare();
-  }
+    return withErrorHandling(
+      this.context,
+      this.emitter,
+      async () => {
+        await this.call?.unmute();
+      },
+      Promise.resolve()
+    );
+  };
 
-  public async stopScreenShare(): Promise<void> {
-    if (this.call?.isScreenSharingOn) await this.handlers.onToggleScreenShare();
-  }
+  public startScreenShare = async (): Promise<void> => {
+    return withErrorHandling(
+      this.context,
+      this.emitter,
+      async () => {
+        if (!this.call?.isScreenSharingOn) await this.handlers.onToggleScreenShare();
+      },
+      Promise.resolve()
+    );
+  };
+
+  public stopScreenShare = async (): Promise<void> => {
+    return withErrorHandling(
+      this.context,
+      this.emitter,
+      async () => {
+        if (this.call?.isScreenSharingOn) await this.handlers.onToggleScreenShare();
+      },
+      Promise.resolve()
+    );
+  };
 
   //TODO: a better way to expose option parameter
-  public startCall(participants: string[]): Call | undefined {
+  public startCall = (participants: string[]): Call | undefined => {
     const idsToAdd = participants.map((participant) => {
       // FIXME: `onStartCall` does not allow a Teams user.
       // Need some way to return an error if a Teams user is provided.
       const backendId = fromFlatCommunicationIdentifier(participant) as CommunicationUserIdentifier;
       return backendId;
     });
 
-    return this.handlers.onStartCall(idsToAdd);
-  }
+    try {
+      return this.handlers.onStartCall(idsToAdd);
+    } catch (e) {
+      this.context.setError(e);
+      this.emitter.emit('error', e);
+      return undefined;
+    }
+  };
 
-  public async removeParticipant(userId: string): Promise<void> {
-    this.handlers.onParticipantRemove(userId);
-  }
+  public removeParticipant = async (userId: string): Promise<void> => {
+    return withErrorHandling(
+      this.context,
+      this.emitter,
+      async () => {
+        await this.handlers.onParticipantRemove(userId);
+      },
+      Promise.resolve()
+    );
+  };
 
-  public getState(): CallAdapterState {
+  public getState = (): CallAdapterState => {
```

Comment:
out of interest, whats the different between these syntaxes?

Reply:
The second style creates an arrow function with the current 'this' variable enclosed. The first style does not. This means if the first style is passed to another function and called, the 'this' variable refers to that other function's 'this' and not the original 'this' of the original adapter.

For example:

`<AnotherComponent getState={adapter.getState} />`

If getState is in first style, calling it inside AnotherComponent will return undefined because 'this.state' refers to AnotherComponent's 'this'. But in the second style the above should work since the 'this' is refering to the adapter's 'this' instead of the AnotherComponent's 'this'.

Reply:
Ah I just naively assumed class methods would retain this as the class instance! Thanks!

Reply:
Is the change to arrow functions necessary for the main thrust of this PR (i.e., hooking up error handling?). Since both changes are both functionally and in SLOC, I'd request separate PRs for the two (land the arrow function first if it's needed for error handling).

OTOH, I'd like to understand the need better.
MDN [discourages the use of arrow functions for object methods](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions): 
![image](https://user-images.githubusercontent.com/82062616/122273625-7e0b0600-ce96-11eb-865a-8edf7de2fd91.png)

While MDN doesn't have anything to say for classes, this [blog post nicely explains](https://javascriptweblog.wordpress.com/2015/11/02/of-classes-and-arrow-functions-a-cautionary-tale/) why it's better to not use arrow functions in classes either: Arrow functions get added as porperties, so (i) they don't show up on the object's prototype chain; (ii) a new copy of the function is created for each instance of the class.

Now, (ii) is unlikely to cause performance issues here, since we don't expect to have many instances of the Adapter. Still, this is an anti-pattern so we should have a really good reason to use arrow functions here, and should add very obvious documentation and caution in the code so that folks aren't surprised down the line.

Reply:
Thanks for the links Prathmesh. The arrow functions are not necessary for this PR and I've removed them from this PR. We may want to decide on which style to use and how users should use the API as we have used different styles in different places.